hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/solr/SolrUtil.java

```diff
@@ -0,0 +1,492 @@
+  private AtomicReference<EntityReadAccessHeatMapResponse>
+      entityReadAccessHeatMapRespRef;
+  private SimpleDateFormat dateFormat = new SimpleDateFormat(
+      "yyyy-MM-dd'T'HH:mm:ss'Z'");
+  private final String timeZone;
+
+  @Inject
+  public SolrUtil(ReconNamespaceSummaryManager
+                      namespaceSummaryManager,
+                  ReconOMMetadataManager omMetadataManager,
+                  OzoneStorageContainerManager reconSCM,
+                  OzoneConfiguration ozoneConfiguration) {
+    this.reconNamespaceSummaryManager = namespaceSummaryManager;
+    this.omMetadataManager = omMetadataManager;
+    this.reconSCM = reconSCM;
+    this.entityReadAccessHeatMapRespRef = new AtomicReference<>(
+        new EntityReadAccessHeatMapResponse());
+    this.ozoneConfiguration = ozoneConfiguration;
+    this.timeZone = this.ozoneConfiguration.get(OZONE_RECON_SOLR_TIMEZONE_KEY,
+        DEFAULT_TIMEZONE_VALUE);
+    if (timeZone != null) {
+      LOG.info("Setting timezone to " + timeZone);
+      try {
+        dateFormat.setTimeZone(TimeZone.getTimeZone(timeZone));
+      } catch (Throwable t) {
+        LOG.error("Error setting timezone. TimeZone = " + timeZone);
+      }
+    }
+  }
+
+  private void addBucketData(
+      EntityReadAccessHeatMapResponse volumeEntity, String[] split,
+      int readAccessCount, long keySize) {
+    List<EntityReadAccessHeatMapResponse> children =
+        volumeEntity.getChildren();
+    EntityReadAccessHeatMapResponse bucketEntity = null;
+    List<EntityReadAccessHeatMapResponse> bucketList =
+        children.stream().filter(entity -> entity.getLabel().
+            equalsIgnoreCase(split[1])).collect(Collectors.toList());
+    if (bucketList.size() > 0) {
+      bucketEntity = bucketList.get(0);
+    }
+    if (children.contains(bucketEntity)) {
+      addPrefixPathInfoToBucket(split, bucketEntity, readAccessCount, keySize);
+    } else {
+      addBucketAndPrefixPath(split, volumeEntity, readAccessCount, keySize);
+    }
+  }
+
+  private void addVolumeData(
+      EntityReadAccessHeatMapResponse rootEntity,
+      String[] split, int readAccessCount, long keySize) {
+    List<EntityReadAccessHeatMapResponse> children =
+        rootEntity.getChildren();
+    EntityReadAccessHeatMapResponse volumeInfo =
+        new EntityReadAccessHeatMapResponse();
+    volumeInfo.setLabel(split[0]);
+    children.add(volumeInfo);
+    addBucketAndPrefixPath(split, volumeInfo, readAccessCount, keySize);
+  }
+
+  private void updateVolumeSize(
+      EntityReadAccessHeatMapResponse volumeInfo) {
+    List<EntityReadAccessHeatMapResponse> children =
+        volumeInfo.getChildren();
+    children.stream().forEach(bucket -> {
+      volumeInfo.setSize(volumeInfo.getSize() + bucket.getSize());
+      updateBucketLevelMinMaxAccessCount(bucket);
+      updateBucketColors(bucket);
+    });
+  }
+
+  private void updateBucketColors(EntityReadAccessHeatMapResponse bucket) {
```

Comment:
Color is mapped by UI, its mostly access ratio, can rename method accordingly

Reply:
Ok. renamed.